third_party/user/user.go

```diff
@@ -378,7 +378,7 @@ func GetExecUser(userSpec string, defaults *ExecUser, passwd, group io.Reader) (
 
 				// Okay, so it's numeric. We can just roll with this.
 			}
-		} else if len(groups) > 0 {
+		} else if len(groups) > 0 && uidErr != nil {
```

Comment:
Exactly, it's to apply to the convert spec. I figured this is a good place to put this logic.